plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java

```diff
@@ -157,7 +157,8 @@ public static boolean isTrinoMaterializedView(String tableType, Map<String, Stri
     public static boolean canDecodeView(Table table)
     {
         // we can decode Hive or Presto view
-        return table.getTableType().equals(VIRTUAL_VIEW.name());
+        return table.getTableType().equals(VIRTUAL_VIEW.name()) &&
+                !isTrinoMaterializedView(table.getTableType(), table.getParameters());
```

Comment:
As I replied offline, I would recommend improving the error message when executing `SELECT` query. The past error message wasn't nice, but the latest error is definitely confusing to users.
```sql
SHOW TABLES IN hive.tpch;
                Table
-------------------------------------
 customer
 lineitem
 nation
 orders
 part
 partsupp
 region
 st_4895c7d3b34545bba114c36d756da40d
 supplier
 test

-- Before this change
SELECT * FROM hive.tpch.test;
Query 20221130_231311_00042_62rkt failed: View data missing prefix: /* Presto Materialized View: eyJvcmlnaW5hbFNxbCI6IlNFTEVDVCAxIGFcblxuIiwiY29sdW1ucyI6W3sibmFtZSI6ImEiLCJ0eXBlIjoiaW50ZWdlciJ9XX0= */

-- After this change
SELECT * FROM hive.tpch.test;
Query 20221130_232023_00030_9azzm failed: Table 'tpch.test' not found
```